lib/enkf/ensemble_config.cpp

```diff
@@ -788,30 +791,26 @@ stringlist_type * ensemble_config_alloc_keylist_from_var_type(const ensemble_con
 
 stringlist_type * ensemble_config_alloc_keylist_from_impl_type(const ensemble_config_type * config , ert_impl_type impl_type) {
   stringlist_type * key_list = stringlist_alloc_new();
-  hash_iter_type * iter = hash_iter_alloc(config->config_nodes);
-  while (!hash_iter_is_complete( iter )) {
-    const char * key = hash_iter_get_next_key(iter);
-    if (enkf_config_node_get_impl_type( (enkf_config_node_type * ) hash_get(config->config_nodes , key)) == impl_type)
+
+  for (const auto& config_pair : config->config_nodes) {
+    const char * key       = config_pair.first.c_str();
+    if (impl_type == enkf_config_node_get_impl_type( config_pair.second ))
       stringlist_append_copy( key_list , key );
 
   }
-  hash_iter_free(iter);
+
   return key_list;
 }
 
 
 bool ensemble_config_has_impl_type(const  ensemble_config_type * config, const ert_impl_type impl_type) {
-  bool ret = false;
-  hash_iter_type * iter = hash_iter_alloc(config->config_nodes);
-  while (!hash_iter_is_complete( iter )) {
-    const char * key = hash_iter_get_next_key(iter);
-    if (enkf_config_node_get_impl_type( (enkf_config_node_type * ) hash_get(config->config_nodes , key)) == impl_type) {
-      ret = true;
-      break;
-    }
+  for (const auto& config_pair : config->config_nodes) {
+    const char * key       = config_pair.first.c_str();
```

Comment:
Does not seem like you use the key for anything?

Reply:
Looks like you are right - fixed.